lib/puppet/functions/mcollective/crontimes.rb

```diff
@@ -0,0 +1,22 @@
+Puppet::Functions.create_function(:'mcollective::crontimes') do
+  dispatch :crontimes do
+    required_param 'Integer', :offset
+    required_param 'Integer', :interval
+    required_param 'Integer', :period
+  end
+
+  def crontimes(offset, interval, period)
+    idx = offset
+    res = [offset]
+    #Loop for period/interval, but if the result is beyond period
+    for i in 1..(period/interval)
+       idx = interval + idx
+       #make sure we do not pass the period we are trying to randomize for
+       if idx >= period
+         break
+       end
+       res << idx
+    end
+    return res
+  end
```

Comment:
But we need to be careul to stay below 60 and need to be careful to avoid 0 in period of interval